postgre-client/src/lib.rs

```diff
@@ -187,4 +188,31 @@ impl PgClient {
             .observe_request(SQL_COMPONENT, TRANSACTION_ACTION, "rollback", start_time);
         Ok(())
     }
+
+    // at the moment used in integration tests only
+    pub async fn clean_db(&self) -> Result<(), IndexDbError> {
+        let mut transaction = self.pool.begin().await?;
+
+        self.drop_indexes(&mut transaction).await?;
+        self.drop_constraints(&mut transaction).await?;
+        for table in [
+            "assets_v3",
+            "assets_authorities",
+            "asset_creators_v3",
+            "batch_mints",
+            "core_fees",
+            "fungible_tokens",
+        ] {
+            self.truncate_table(&mut transaction, table).await?;
+        }
+
+        transaction.execute(sqlx::query("update last_synced_key set last_synced_asset_update_key = null where id = 1;")).await?;
+
+        self.recreate_indexes(&mut transaction).await?;
+        self.recreate_constraints(&mut transaction).await?;
+
+        transaction.commit().await.map_err(|e| e)?;
```

Comment:
fyi, those await? above will not always rollback the tx, a minor concern if its only in tests

Reply:
added comment, thanks for info